contracts/modules/OptimismKeyStoreProofModule/KnownStateRootWithHistory.sol

```diff
@@ -0,0 +1,77 @@
+pragma solidity ^0.8.17;
+
+import "./IKnownStateRootWithHistory.sol";
+import "./IL1Block.sol";
+import "./BlockVerifier.sol";
+
+contract KnownStateRootWithHistory is IKnownStateRootWithHistory {
+    mapping(uint256 => BlockInfo) public stateRoots;
+    uint256 public constant ROOT_HISTORY_SIZE = 10;
+    uint256 public currentRootIndex = 0;
+    // https://community.optimism.io/docs/developers/build/differences/#accessing-l1-information
+    address public immutable L1_BLOCK;
```

Comment:
change `address L1_BLOCK` to `IL1Block L1_BLOCK` make others easier to understand